### ISSUE TITLE
Nice work! - I want to use for searching parts within a song....

### ISSUE THREAD

**Issue by johndpope** [ztjordan/ShazamClone#1]
If I highlight / select a search range of .5 seconds at 6 seconds - can I search within a single song to find all the matching transients.

![image](https://user-images.githubusercontent.com/289994/142733310-d80efb3c-e8b3-4bcb-8823-8a5d8d7f5480.png)

I basically want to make a clone of this for desktop
https://www.softpedia.com/get/Multimedia/Audio/Other-AUDIO-Tools/FindSounds-Palette.shtml

https://www.findsounds.com/palette.html

![image](https://user-images.githubusercontent.com/289994/142733392-aef638f0-b242-49f4-a2c2-aef0174375e5.png)

don't need a gui - just capability / api to search

**Comment by ztjordan** [comment]
Hey John - I want to make sure I understand this correctly. You want to find a snippet inside of a song and then find all of the snippets inside of the same song (or possibly a different song) that match?

**Comment by johndpope** [comment]
yeah - spot on. like find a kick - and then tell me all the spots / times in that song that have said kick. 
It's easier when you can see / as per findsounds - they highlight the colours ( you can use like an eye drop picker from photoshop to select) - but maybe - it could done using existing spread of signatures to create  a colour / sound palette. 


If I could dig into this data spreadspread - that would be great. 
1) give song -> get all the fingerprints / and number of uses.
eg. from given song - there's 1,234 unique finger prints. 
for fingerprint A - there's 117 times -
 for fingerprint B - there's 10 times use etc. 
 
2) given a fingerprint / return array of all times used.....

eg.
fingerprint A
[0:01:20 - 0:01:24, 0:02:24 ,...]

presumably these 2 points sound alike....

if we can get this kind of data - could open up an entire research into unsurpervised learning of song deconstruction.
like - imagine I have a 100 mbs of kicks on my computer - we can index all that - then run a song against it - and find exact matches. - then potentially can rebuild the drum section using AI (phase 3 or 4).

I don't need to store this data permanently in the database for entire song I'm working on. I get shazam compressed data / chucks stuff out.  Just want to query it - maybe stash in a temp table.
